tools/install/hcledit.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2155 # No way to assign to readonly variable in separate lines
+readonly SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd -P)"
+# shellcheck source=_common.sh
+. "$SCRIPT_DIR/_common.sh"
+
+#
+# Unique part
+#
+
+GH_ORG="minamijoyo"
+GH_RELEASE_REGEX_LATEST="https://.+?_${TARGETOS}_${TARGETARCH}.tar.gz"
+GH_RELEASE_REGEX_SPECIFIC_VERSION="https://.+?${VERSION}_${TARGETOS}_${TARGETARCH}.tar.gz"
+DISTRIBUTED_AS="tar.gz"
```

Comment:
Maybe re-use `DISTRIBUTED_AS`?
```suggestion
DISTRIBUTED_AS="tar.gz"
GH_RELEASE_REGEX_LATEST="https://.+?_${TARGETOS}_${TARGETARCH}.${DISTRIBUTED_AS}"
GH_RELEASE_REGEX_SPECIFIC_VERSION="https://.+?${VERSION}_${TARGETOS}_${TARGETARCH}.${DISTRIBUTED_AS}"
```

Reply:
Nope, because:
1. It will be different for binary
2. You will know the Distribution type only when you get GH Release link, not otherwise

Reply:
> 1. It will be different for binary

The `binary` types just don't use the `DISTRIBUTED_AS` in `GH_RELEASE_REGEX_LATEST` and `GH_RELEASE_REGEX_SPECIFIC_VERSION` vars so they are not affected at all by the suggested change.


> 2. You will know the Distribution type only when you get GH Release link, not otherwise

This is probably about something else.
I see the code:
```shell
GH_RELEASE_REGEX_LATEST="https://.+?_${TARGETOS}_${TARGETARCH}.tar.gz"
GH_RELEASE_REGEX_SPECIFIC_VERSION="https://.+?${VERSION}_${TARGETOS}_${TARGETARCH}.tar.gz"
DISTRIBUTED_AS="tar.gz"
```
and
```shell
GH_RELEASE_REGEX_LATEST="https://.+?/${TOOL}_${TARGETOS}_${TARGETARCH}"
GH_RELEASE_REGEX_SPECIFIC_VERSION="https://.+?v${VERSION}/${TOOL}_${TARGETOS}_${TARGETARCH}"
DISTRIBUTED_AS="binary"
```
and I see no reason to not swap these two to the below:
```shell
DISTRIBUTED_AS="tar.gz"
GH_RELEASE_REGEX_LATEST="https://.+?_${TARGETOS}_${TARGETARCH}.${DISTRIBUTED_AS}"
GH_RELEASE_REGEX_SPECIFIC_VERSION="https://.+?${VERSION}_${TARGETOS}_${TARGETARCH}.${DISTRIBUTED_AS}"
```
and
```shell
DISTRIBUTED_AS="binary"
GH_RELEASE_REGEX_LATEST="https://.+?/${TOOL}_${TARGETOS}_${TARGETARCH}"
GH_RELEASE_REGEX_SPECIFIC_VERSION="https://.+?v${VERSION}/${TOOL}_${TARGETOS}_${TARGETARCH}"
```

No change to `binary` types and optimization to `non-binary` types.

Am I missing something non-obvious?

Reply:
Once again:

> You will know the Distribution type only when you get GH Release link, not otherwise

Imagine that you adding a new hook, let's say tfupdate.

You copy-past file, IE `tfsec.sh` rename it, cleanup pre-populated values and get this:

```bash
#...

GH_ORG=""
GH_RELEASE_REGEX_LATEST="https://.+?/"
GH_RELEASE_REGEX_SPECIFIC_VERSION="https://.+?"
DISTRIBUTED_AS=""

common::install_from_gh_release "$GH_ORG" "$DISTRIBUTED_AS" \
  "$GH_RELEASE_REGEX_LATEST" "$GH_RELEASE_REGEX_SPECIFIC_VERSION"
```

How you should know `DISTRIBUTED_AS` value till you don't get any of `GH_RELEASE_...`?

Please, don't make it more complex than it already is.

Also, someone could distribute not `.tar.gz` but `.tgz`.

Reply:
> How you should know `DISTRIBUTED_AS` value till you don't get any of `GH_RELEASE_...`?

From what I see in each new script for each tool installation, the value of `DISTRIBUTED_AS` is known and is defined in each of these files. Along with that most of the files have the value of `DISTRIBUTED_AS` repeated **three** times. And my suggestion is to reduce it to **one** time.


> Please, don't make it more complex than it already is.

I'm trying to simplify so that new tools are added without a need to copy&paste `DISTRIBUTED_AS` value **three** times on a three lines in a row.

![image](https://github.com/antonbabenko/pre-commit-terraform/assets/1274789/85ab053a-88df-4668-beb7-fb5508f8258c)

Reply:
There 0 simplification.

Reply:
Read once again, maybe tomorrow
https://github.com/antonbabenko/pre-commit-terraform/pull/644#discussion_r1529207484

Reply:
There is a process of adding a new tool from scratch
[Screencast from 24.04.24 21:01:35.webm](https://github.com/antonbabenko/pre-commit-terraform/assets/11096782/19e679c4-8bd7-4e5c-84a2-c3ab9551fa79)

Reply:
1. Get URLs
2. Replace URL parts with variables
3. Simplify URL by replacing not matter parts by regex
4. Set `DISTRIBUTED_AS` value based on URL

Also, I see that I'm already stuck to recreate that process from memory for LATEST regex, which means that this flow is already complicated. 
That verbosity does not hurt anyone.

Reply:
Version without stuck (will add it to contributor docs)
[Screencast from 24.04.24 21:28:39.webm](https://github.com/antonbabenko/pre-commit-terraform/assets/11096782/8fb13b31-1aff-44b7-a642-27f2fce3477f)

Reply:
Version without stuck (will add it to contributor docs)

[Screencast from 24.04.24 22:00:36.webm](https://github.com/antonbabenko/pre-commit-terraform/assets/11096782/8fc461e9-f163-4592-9497-4a18fa89c0e8)